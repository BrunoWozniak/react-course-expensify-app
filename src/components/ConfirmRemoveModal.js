import React from 'react';
import Modal from 'react-modal';

const ConfirmRemoveModal = (props) => (
    <Modal
        isOpen={!!props.confirmDelete}
        contentLabel="Are you sure?"
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">Are you sure?</h3>
        <p className="modal__body">Please confirm you want to delete this expense.</p>
        <div className="modal__buttons">
            <button className="button" name="removeConfirmButton" onClick={props.onConfirmDelete}> Confirm </button>
            <button className="button button--secondary" name="removeCancelButton"onClick={props.onCancelDelete} > Cancel </button>
        </div>
    </Modal>
);

export default ConfirmRemoveModal;