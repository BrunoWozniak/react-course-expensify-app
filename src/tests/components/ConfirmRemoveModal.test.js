import React from 'react';
import { shallow } from 'enzyme';
import ConfirmRemoveModal from '../../components/ConfirmRemoveModal';

let startRemoveExpense, history, wrapper;

beforeEach(() => {
    history = { push: jest.fn() };
    wrapper = shallow(<ConfirmRemoveModal
        confirmDelete={true}
        history={history}
    />);
});

test('should render ConfirmRemoveModal correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

// Missing: 'should handle onCancel' and 'should handle onConfirm'
// test('should handle onCancel', () => {
//     wrapper.find({ name: 'removeCancelButton' }).simulate('click');
//     expect(wrapper.state('confirmDelete')).toBe(false);
// });