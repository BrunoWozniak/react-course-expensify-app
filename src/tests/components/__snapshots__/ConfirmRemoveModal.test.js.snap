// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfirmRemoveModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Are you sure?"
  isOpen={true}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
>
  <h3
    className="modal__title"
  >
    Are you sure?
  </h3>
  <p
    className="modal__body"
  >
    Please confirm you want to delete this expense.
  </p>
  <div
    className="modal__buttons"
  >
    <button
      className="button"
      name="removeConfirmButton"
    >
       Confirm 
    </button>
    <button
      className="button button--secondary"
      name="removeCancelButton"
    >
       Cancel 
    </button>
  </div>
</Modal>
`;
